### PR TITLE
test/e2e: move some tests from cluster-lifecycle to cloud-provider-gcp

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

Move GKE/GCE specific tests from the test/e2e/lifecycle to a new gcp package.
Ideally these tests should be owned by the GCP subproject of sig-cloud-provider.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/kind cleanup
/area test
/priority important-longterm
/sig release cloud-provider cluster-lifecycle testing
/assign @andrewsykim @timothysc 
/cc @justaugustus @timothysc @justinsb @alejandrox1

### CODE REVIEW

File: test/test_owners.csv
Comment (neolit123) [inline test/test_owners.csv]: some of these test strings are MIA, is this file even used today?

**Comment by neolit123** [comment]
xref https://github.com/kubernetes/test-infra/pull/14701

**Review by alejandrox1 — COMMENTED** [review]
Thank you for going through these!
/lgtm

File: test/e2e/cloud/cluster_upgrade.go
Comment (andrewsykim) [inline test/e2e/cloud/cluster_upgrade.go]: Upgrade tests seem like cluster lifecycle to me, are we moving because the test is GCP specific?

Comment (alejandrox1) [reply test/e2e/cloud/cluster_upgrade.go]: I believe something that we should mention here is that a lot of these tests rely on the cluster directory stuff. We have been moving a lot of the unmaintained things that rely on cluster/ to sig-gcp (dormant group) because some of the prominent parts of cluster/ are gcp specific

**Comment by alejandrox1** [comment]
/test pull-kubernetes-verify
this thing has been flaking a lot https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83744/pull-kubernetes-verify/1182343462616305664

File: test/e2e/cloud/cluster_upgrade.go
Comment (neolit123) [reply test/e2e/cloud/cluster_upgrade.go]: > Upgrade tests seem like cluster lifecycle to me, are we moving because the test is GCP specific?

yes, apart from @justinsb nobody in the SIG knows how to debug these.

**Comment by neolit123** [comment]
@alejandrox1 
> this thing has been flaking a lot https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83744/pull-kubernetes-verify/1182343462616305664

> Errors from staticcheck:
test/e2e/cloud/cluster_upgrade.go:445:2: argument upgCtx is overwritten before first use (SA4009)

looks like a real problem, but no clear why it wasn't present before.

**Review by timothysc — APPROVED** [review]
Loong overdue. 
/approve 
/assign @justinsb

**Comment by andrewsykim** [comment]
/hold 

(I have some feedback pending)

File: test/e2e/cloud/cluster_upgrade.go
Comment (andrewsykim) [inline test/e2e/cloud/cluster_upgrade.go]: For the tests we know for sure are gcp specific, can we bundle them together into `gcp.go` file or a `gcp` package - `test/e2e/cloud/gcp/`?

Comment (neolit123) [reply test/e2e/cloud/cluster_upgrade.go]: i'm +1 on doing that.
a couple of questions would be
- how to name the SIG tag? `sig-cloud-provider-gcp` SGTM
- who to add in the OWNERS files possibly @justinsb and @cheftako

Comment (justaugustus) [reply test/e2e/cloud/cluster_upgrade.go]: Yep, `sig-cloud-provider-gcp` is how we're tagging in test-infra FWIW.

File: test/test_owners.csv
Comment (justaugustus) [inline test/test_owners.csv]:
```suggestion
Upgrade cluster upgrade should maintain a functioning cluster,justinsb,1,cloud-provider
```

File: test/test_owners.csv
Comment (justaugustus) [inline test/test_owners.csv]:
```suggestion
Upgrade master upgrade should maintain a functioning cluster,justinsb,1,cloud-provider
```

File: test/e2e/cloud/cluster_upgrade.go
Comment (neolit123) [reply test/e2e/cloud/cluster_upgrade.go]: ok, i'm going to introduce `//gcp` with the OWNERS and this tag.

File: test/test_owners.csv
Comment (justaugustus) [inline test/test_owners.csv]:
```suggestion
Upgrade node upgrade should maintain a functioning cluster,justinsb,1,cloud-provider
```

**Comment by neolit123** [comment]
@andrewsykim i kept the tests in multiple files because they are better scoped this way.
a single gcp.go ends being quite big.

File: test/e2e/gcp/cluster_upgrade.go
Comment (cheftako) [inline test/e2e/gcp/cluster_upgrade.go]: No such sig. I think its either sig-cloud-provider or the appropriate GCP WG.

Comment (neolit123) [reply test/e2e/gcp/cluster_upgrade.go]: as per @justaugustus 's comment on slack:
```
It's a subproject, so if we're being super correct, it's sig-cloud-provider-provider-gcp, but that's shortened to sig-cloud-provider-gcp in test-infra
```
https://app.slack.com/team/U0E0E78AK

so in a way this current tag is valid for the sub-project.

**Comment by andrewsykim** [comment]
@neolit123 can we move `test/e2e/gcp` -> `test/e2e/cloud/gcp`?

**Comment by neolit123** [comment]
@andrewsykim 

> @neolit123 can we move test/e2e/gcp -> test/e2e/cloud/gcp?

SGTM, will update in a bit.

**Comment by neolit123** [comment]
/assign @BenTheElder 

for the staticcheck / hack update.

**Comment by neolit123** [comment]
@andrewsykim 

> can we move test/e2e/gcp -> test/e2e/cloud/gcp?

done

File: test/e2e/cloud/gcp/cluster_upgrade.go
Comment (BenTheElder) [inline test/e2e/cloud/gcp/cluster_upgrade.go]: is this the right tag now? /shrug

Comment (neolit123) [reply test/e2e/cloud/gcp/cluster_upgrade.go]: we didn't want `sig-cloud-provider` and gcp is a sub-project so there is no valid tag for this
https://github.com/kubernetes/kubernetes/pull/83744#discussion_r333694270

Comment (BenTheElder) [reply test/e2e/cloud/gcp/cluster_upgrade.go]: ok, just not sure what the pattern is, I'll drop a /approve and let someone else bikeshed ;-)

File: test/e2e/cloud/gcp/OWNERS
Comment (BenTheElder) [inline test/e2e/cloud/gcp/OWNERS]: cc @justinsb @cheftako

**Comment by BenTheElder** [comment]
/approve
for hack

File: test/e2e/cloud/gcp/cluster_upgrade.go
Comment (neolit123) [reply test/e2e/cloud/gcp/cluster_upgrade.go]: these are mostly for the visibility of the release team and triage.

**Comment by alejandrox1** [comment]
/lgtm
@andrewsykim could you take another look, please?

**Comment by neolit123** [comment]
merging on lazy consensus later this week, if there are no objections.

**Review by cheftako — COMMENTED** [review]
/lgtm

**Comment by justaugustus** [comment]
No objections from me!

**Comment by timothysc** [comment]
/hold cancel
^YaY!

File: test/test_owners.json
Comment (andrewsykim) [inline test/test_owners.json]: Is this still right?

Comment (andrewsykim) [reply test/test_owners.json]: Ah I see, this is based on test/test_owners.csv

**Comment by andrewsykim** [comment]
/lgtm

Thanks @neolit123

File: test/test_owners.json
Comment (neolit123) [reply test/test_owners.json]: looks like test_owners* is tech debt that is bound to both k/k and test-infra ATM.

**Comment by spiffxp** [comment]
This dropped tests from the e2e.test binary, I've opened #85962 to address

**Pull request merged**
